src/go/k8s/tests/e2e/redpanda-schema-registry/00-redpanda-cluster.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: redpanda.vectorized.io/v1alpha1
+kind: Cluster
+metadata:
+  name: schema-registry
+spec:
+  image: "vectorized/redpanda"
+  version: "latest"
+  replicas: 1
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+    limits:
+      cpu: 1
+      memory: 100Mi
+  configuration:
+    rpcServer:
+      port: 33145
+    kafkaApi:
+    - port: 9092
+      tls:
+        enabled: true
+    adminApi:
+    - port: 9644
+    schemaRegistry:
+      port: 8081
```

Comment:
nit: could you drop the port to also test the default one?

Reply:
np

Reply:
why did this move back? It's not possible to use the default? 🤔

Reply:
I moved this back because if I remove `port` the `schemaRegistry` is alone and I would need to do this like:
```yaml
spec:
  configuration:
    schemaRegistry: {}
```

In my opinion this is misleading.

I put default port because when user look at our different examples:
```yaml
spec:
  configuration:
    kafkaAPI:
    - port: 9999
    - external:
        enabled: true
```

I would envision that they might want to copy this to the schema registry configuration. The result would be 
```yaml
spec:
  configuration:
    schemaRegistry:
      external:
        enabled: true
```

Which is in my opinion valid, but I would like to not do defaults inside `ClusterServiceResource` nor in `StatefulsetResource`. If we are using webhooks this is, for me, right place to put  defaults values.

I could split `redpanda-schema-registry` e2e to have 3 different scenarios, but I wanted to verify that our operator can correctly change the resource while Redpanda is running. 

There is still issue with transition between schema registry without external connectivity but with TLS certificates to the schema registry with external connectivity with TLS certificates. As far as I remember cert manager didn't react on the change in `Certificate`, because secrets are available.

Reply:
makes sense, thanks